[SkinnyClient] Add Skinny Client initial build. Still has numpy, pandas and sqlparse dependency.

eddiestudies opened the pull request:
## What changes are proposed in this pull request?

Walks through process of removing dependencies and having test_client.py pass

## How is this patch tested?

(Details)

## Release Notes

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

Commit:
Save dependency removals for tracking

Commit:
Save dependency removals for tracking

Commit:
Add fixes for test_client.py

Commit:
Add fixes for test_client.py

eedeleon commented on mlflow/fastai.py:
```diff
@@ -17,8 +17,6 @@
 import gorilla
 import tempfile
 import shutil
-import pandas as pd
```
pandas and numpy dependencies in flavors should be lazy loaded https://github.com/mlflow/mlflow/issues/3577

eedeleon commented on mlflow/store/artifact/databricks_artifact_repo.py:
```diff
@@ -151,6 +148,8 @@ def _azure_upload_file(self, credentials, local_file, artifact_path):
         Finally, since the prevailing credentials could expire in the time between the last
         stage_block and the commit, a second try-except block refreshes credentials if needed.
         """
+        from azure.core.exceptions import ClientAuthenticationError
```
This would mean the mlflow base library may not work for databricks scenarios by default.

eedeleon commented on mlflow/store/tracking/file_store.py:
```diff
@@ -720,6 +718,7 @@ def _list_run_infos(self, experiment_id, view_type):
     def _search_runs(
         self, experiment_ids, filter_string, run_view_type, max_results, order_by, page_token
     ):
+        from mlflow.utils.search_utils import SearchUtils
```
MLflow SearchRuns depends on pandas, numpy, and sqlparse. How should these dependencies be managed for skinny client?

eedeleon commented on setup.py:
```diff
@@ -31,32 +31,19 @@ def package_files(directory):
-        "sqlparse>=0.3.1",
-        # Required to run the MLflow server against SQL-backed storage
-        "sqlalchemy<=1.3.13",
-        "gorilla",
```
For skinny client to be shipped, we would need to update the setup.py to be configurable. Alongside deployment scripts to leverage the configuration https://github.com/mlflow/mlflow/issues/3579

dbczumar commented on setup.py:
```diff
@@ -31,32 +31,19 @@ def package_files(directory):
-        'waitress; platform_system == "Windows"',
-        'gunicorn; platform_system != "Windows"',
-        "Flask",
-        "numpy",
-        "pandas",
```
I noticed a lot of attention devoted to `numpy`, `pandas`, and `sqlparse` throughout other portions of this skinny client refactoring. Is it absolutely necessary to exclude these libraries? They aren't currently versioned and are fundamental to most data science / ML workflows anyway. Have we had any higher-level conversations or done any design regarding what should and should not be included in a skinny client?

iclrdpsynthdataeval replied:
I do not think it is absolutely necessary.  sqlparse, if it is a small dependency could be included back to enable search_runs support. I think there was a general push to have the DS specific libraries removed, however, as you noticed from the PR, even lazy loading those packages involves a lot of code changes.

eedeleon replied:
I do not think it is absolutely necessary. sqlparse, if it is a small dependency could be included back to enable search_runs support. I think there was a general push to have the DS specific libraries removed, however, as you noticed from the PR, even lazy loading those packages involves a lot of code changes.

dbczumar replied:
Got it - can we put together a list of requirements for the skinny client that will help us make decisions on which libraries are / are not necessary? (I.e. it'd be awesome to understand what the general push looks like). I think @akshaya-a had done some work here? Would be excited to meet and discuss this as well!

dbczumar commented on setup.py:
```diff
@@ -31,32 +31,19 @@ def package_files(directory):
-        "python-dateutil",
-        "protobuf>=3.6.0",
-        "gitpython>=2.1.0",
-        "pyyaml",
```
Required for running MLflow Projects, saving MLflow models

eedeleon commented on setup.py:
```diff
@@ -31,32 +31,19 @@ def package_files(directory):
-        # Pin sqlparse for: https://github.com/mlflow/mlflow/issues/3433
-        "sqlparse>=0.3.1",
-        # Required to run the MLflow server against SQL-backed storage
-        "sqlalchemy<=1.3.13",
```
_tracking_store_registry.register(scheme, _get_sqlalchemy_store) should be called only if sqlalchemy is available

eedeleon commented on setup.py:
```diff
@@ -31,32 +31,19 @@ def package_files(directory):
     install_requires=[
-        "alembic<=1.4.1",
-        # Required
-        "azure-storage-blob",
```
Artifact repositories should also be lazy loaded based off of dependencies https://github.com/mlflow/mlflow/blob/master/mlflow/store/artifact/artifact_repository_registry.py

eedeleon commented on tests/tracking/test_model_registry_client.py:
```diff
@@ -0,0 +1,97 @@
+import pytest
+from unittest import mock
```
As this test split suggests, we will need to define a test split for mlflow vs mlflow-skinny

andrewnitu committed:
Add universal mlflow.autolog() function! (#3561)

* stash

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* commit import test

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* stash

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* stash

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* idk

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* stash changes

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* i think this works...

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* remove logging msg

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* revert some changes

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* revert blank line

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* commit changes

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* stash

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* stash

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* this is the best i could do

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* add some comments

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* fix linting

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* mark test as large

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* move universal autolog to fluent module

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* pls work

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* remove assert that autolog was not called before module was imported

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* revert signature entity change

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* maybe works?

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* stash

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* working kinda?

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* handle where session exists but mlflow-spark jar is not attached'

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* stash

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* stash

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* finally working i think?

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* Revert changes to spark testing script

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* fix missing jar test

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* attempt to change test to capture logging msg

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* revert changes to testing script

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* revert new pom

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* revert changes to pom.xml

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* revert print

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* fmt all

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* fix remaining tests

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* fmt

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* retrigger tests

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* revert spark example

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* code review

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* more code review

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* fmt

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* temp example change

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* improve tests

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* add test to ensure a throwing specific autolog does not cause mlflow.autolog to throw

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* formatting

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* tests working?

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* working now

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* small fixes and reverts

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* add log messages for when autologging is enabled for a specific library

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* decrement pytest-mock to hopefully work with our old pytest??

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* do better job of cleaning up imports both before and after autolog tests

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* do better job of cleaning up imports both before and after autolog tests

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* move resetting of global states to a fixture

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* pytest-mock older version somehow broke our tests

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* stash

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* cleanup

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* stop using pytest-mock

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* remove references to mocker

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* retrigger tests

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* potential fix for import error in tests

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* fix tests plssss

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* add libraries to small-requirements

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* tests pls

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* bump test versino of pyspark to 3.0.0

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* update test comments

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* test moving ml libraries to large-requirements

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* move universal autolog tests to different file and mark as large

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* move autologging enabled message for spark to only show when a spark session exists or is created

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* try to fix tests??

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* Safe shutdown

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>
Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* Wrapt

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>
Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* Wrapt

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>
Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* Docs

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* Format

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* retrigger tests

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* retrigger tests

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* Remove string "databricks" from comment

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* fix fluent autolog tests

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* assert resetting the state and fix inconsistent dict

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* fix pytest linting

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* fix pycodestyle lint errors

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* do not ignore fluent_autolog tests

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* fix test config hopefully

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* assert that fluent_autolog test IS being run

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* revert 'assert that fluent_autolog test IS being run'

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

Co-authored-by: Corey Zumar <corey.zumar@databricks.com>
Co-authored-by: dbczumar <39497902+dbczumar@users.noreply.github.com>

harupy committed:
Fix a bug where `xgboost.DMatrix(None)` fails after running `mlflow.xgboost.autolog` (#3584)

* Add failing test

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Fix __init__

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

mateiz committed:
Update setup.py (#3591)

Signed-off-by: Matei Zaharia <matei.zaharia@gmail.com>

harupy committed:
Add examples for shap integration (#3542)

* Add shap example

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* fix example in docstring

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Update test_examples.py

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* introduce utils

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Add example for multiclass classsifcation

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* rename regression script

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* fix test

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* use predict_proba

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Add binary classification example

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* update test

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* rename util function

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Add docstring

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* nit

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Fix docstring

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Add readme

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* show create plot instead of summary plot

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* remove unused import

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Update examples/shap/README.md

Co-authored-by: Siddharth Murching <smurching@gmail.com>
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Add command to install required packages

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* nit

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* update readme

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

Co-authored-by: Siddharth Murching <smurching@gmail.com>

AxelVivien25 committed:
Add mlflow-elasticsearchstore to the doc (#3462)

* Add mlflow-elasticsearchstore to the doc

Signed-off-by: Axel Vivien <a.vivien@criteo.com>

* add PyPI link

Signed-off-by: Axel Vivien <a.vivien@criteo.com>

Co-authored-by: Axel Vivien <a.vivien@criteo.com>

harupy committed:
Fix a bug where `lightgbm.Dataset(None)` fails after running `mlflow.lightgbm.autolog` (#3594)

* Add failing test

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Fix __init__

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

dmatrix committed:
Final part of code snippets for fluent and MlflowClient APIs (#3573)

* Added search_runs() and get_metric_history() code snippets

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* Added list_registered_models and create_registered_model code snippet

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* Added delete_registered_model code snippet

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* Added delete_registered_model_tag() code snippet

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* Added get_registered_model() and set_registered_model_tag() code snippet

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* Added create_model_version() code snippet

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* Fixed build problem; added create_model_version() and delete_model_version() code snippet

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* Added more code snippets

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* Last bits of code snippets. Finished now

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* Modified and shortened some snippets

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* Fixed lint errors exceeding max line size

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* fixed minor indent issues; int(mv.version) not needed for some methods

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* include step argument to the get_metirc_history() method

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* Print step parameter in get_metric_history()

Signed-off-by: Jules Damji <dmatrix@comcast.net>

* fixed a minor indent alighment

Signed-off-by: Jules Damji <dmatrix@comcast.net>

harupy committed:
Un-max-pin sqlalchemy (#3598)

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

lorenzwalthert committed:
Fix print method for pyfunc flavor models without run_id and artifact path (#3589)

* add test to make repr method fail

Signed-off-by: Lorenz Walthert <lorenz.walthert@icloud.com>

* only check run_id if attribute exists

Signed-off-by: Lorenz Walthert <lorenz.walthert@icloud.com>

* also check artifact path

Signed-off-by: Lorenz Walthert <lorenz.walthert@icloud.com>

* specify artifact_path and run_id explicitly in Model(), although the same as the defaults

Signed-off-by: Lorenz Walthert <lorenz.walthert@icloud.com>

* use .__repr__() instead of str()

Co-authored-by: Harutaka Kawamura <hkawamura0130@gmail.com>
Signed-off-by: Lorenz Walthert <lorenz.walthert@icloud.com>

Co-authored-by: Harutaka Kawamura <hkawamura0130@gmail.com>

dbczumar committed:
Sklearn autologging: Fix behavior when a child and its parent are both patched (#3582)

* Clazz

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Inline gorilla

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Some docs

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Swap gorilla

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Add test

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Test tweak

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Doc

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* whitespace

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Format with black

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Overwrite option

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Doc

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Revert "Doc"

This reverts commit 5a84bba2002c1dc312359f22cd50e3a6a5b770fd.

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Revert "Overwrite option"

This reverts commit b4fcfa5d1f4ed9c7441ca66bc132310ed37ff9e1.

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Safety

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Black format

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Lint, address comment

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

dbczumar committed:
Scikit-learn autologging: Exclude feature extraction / selection estimator (#3600)

* Exclude feature extraction / selection

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Black

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Drop gorilla

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

andrewnitu committed:
Universal mlflow.autolog() documentation (#3590)

* start docs

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* add docs for function

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* update

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* add reference to tracking docs

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* fix lint

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* Change every -> each

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

* pluralize

Signed-off-by: Andrew Nitu <andrewnitu@gmail.com>

dbczumar committed:
unpin torch (#3603)

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

dbczumar committed:
Unpin most ML dependencies (#3605)

* Many

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Re-pin problematic

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Update extra-ml-requirements.txt

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

shrinath-suresh committed:
Adding support for logging additional artifacts when using mfllow pytorch log and save model (#3436)

dbczumar committed:
Add docs clarifications for pyfunc flavor in mlflow.sklearn (#3609)

* Doc

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* More doc

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Docs

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Further doc

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Punct

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Consistency

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

sephib committed:
Fix keras 2.4 autolog#3571 (#3611)

* Fix bug - when running Keras 2.4.x

Signed-off-by: sephi berry <js.berry@gmail.com>

* fix black

Signed-off-by: sephi berry <js.berry@gmail.com>

sueann committed:
Clean up model registry endpoint and client method definitions (#3610)

Model registry: remove experimental labels, add non-preview REST API paths

Making sure the REST API docs are updated properly will be done in another PR.

Commit:
Move imports in fluent

Commit:
Move imports in fluent

harupy committed:
Remove attrdict from dependencies (#3534)

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

harupy committed:
Pin h5py < 3 to fix incompatibility with TensorFlow model serialization (#3616)

shrinath-suresh committed:
Logging/Loading TorchScripted model (#3557)

shrinath-suresh committed:
mlflow deployments - Adding 'predict' cli command (#3597)

shrinath-suresh committed:
Autolog with PyTorch using lightning (#3601)

harupy committed:
Log torchserve artifacts under `torchserve_artifacts` and log their locations in MLmodel file (#3607)

* log artifact info in MLmodel

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* fix

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* use not equal

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* fix requirements_file path

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Fix test for requirements.txt

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Fix test_requirement_file_save_model

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Fix test_extra_files_log_model

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Fix test_extra_files_save_model

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Assert uri is correct

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* use content_expected

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Use requirements

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* nit

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* nit

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* remove uir

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* fix lint error

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Store files under model

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

* Change constant variable names

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

dbczumar committed:
Autologging: Pluralize configuration arguments (#3617)

* Merge master, fix conflicts

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Fix

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

* Whitespace

Signed-off-by: Corey Zumar <corey.zumar@databricks.com>

shrinath-suresh committed:
Autolog with PyTorch using lightning - Examples (#3608)

Commit:
Remove optimizer step override to ensure that all accelerator cases are covered by base module (#3635)

Signed-off-by: SeanNaren <sean@grid.ai>

shrinath-suresh committed:
Pytorch Autolog - Using tmpdir for model checkpoints and adding updating gitignore (#3632)

* Autolog for pytorch lightning with examples and unit tests

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Adding pytorch lightning 1.0.2 to requirements

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Moving examples to new PR - https://github.com/mlflow/mlflow/pull/3608

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* DDP environment fix - decorating autolog function with rank zero to avoid duplicate runs in multi gpus environment

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Addressin review comments

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Updating the strings

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Updating all key names to lower case

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Fixing Unit Tests

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Adding wrapper for autolog library

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Using wrap patch

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Updating docstring to mention support of pytorch lightning

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Adding more optimizer parameters

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Fixing UT - test_pytorch_autolog_persists_manually_created_run

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Removing unwanted params logged condition check in UT

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Type casting best score to float

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Addressing review comments

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Updating doc string

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Setting autolog method as experimental

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Adding lightning_logs to gitignore and using temp dir to save ckpt files in tests

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Appying black - to fix lint error

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

* Removing Unused import

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

karthik-77 committed:
Mlflow torchserve plugin doc (#3634)

Commit:
Add numpy and pandas, separate tess/tracking/service/ tests for skinny client initial tests. Override tracking uri fixture for service/ tests

Commit:
Add numpy and pandas, separate tess/tracking/service/ tests for skinny client initial tests. Override tracking uri fixture for service/ tests

Commit:
Update test running file

Commit:
Update test running file

Commit:
Remove changes to flavors

Commit:
Remove changes to flavors

Commit:
Merge branch 'eedeleon/skinny_client' of https://github.com/eedeleon/mlflow into eedeleon/skinny_client

eedeleon commented:
I need to update files to upstream/master not origin/master. All flavor files should not be edited. Addressing that now. Do not start review yet
